runbot_travis2docker/models/runbot_build.py

```diff
@@ -0,0 +1,283 @@
+                        " TESTS='1'" in df_content:
+                    build.dockerfile_path = path_script
+                    build.docker_image = build._get_docker_image()
+                    build.docker_container = build._get_docker_container()
+                    if build in to_be_skipped_ids:
+                        to_be_skipped_ids -= build
+                    break
+        if to_be_skipped_ids:
+            _logger.info('Dockerfile without TESTS=1 env. '
+                         'Skipping builds %s', to_be_skipped_ids.ids)
+            to_be_skipped_ids._skip()
+
+    def _local_cleanup(self):
+        builds = self.filtered('branch_id.repo_id.is_travis2docker_build')
+        super(RunbotBuild, self - builds)._local_cleanup()
+        for build in builds:
+            if build.docker_container:
+                subprocess.call(['docker', 'rm', '-f', build.docker_container])
+                subprocess.call(['docker', 'rmi', '-f', build.docker_image])
+
+    def _get_ssh_keys(self):
+        self.ensure_one()
+        response = self.repo_id._github(
+            "/repos/:owner/:repo/commits/%s" % self.name, ignore_errors=True)
+        if not response:
+            return
+        keys = ""
+        for own_key in ['author', 'committer']:
+            try:
+                ssh_rsa = self.repo_id._github('/users/%(login)s/keys' %
+                                               response[own_key])
+                keys += '\n' + '\n'.join(rsa['key'] for rsa in ssh_rsa)
+            except (TypeError, KeyError, requests.RequestException):
+                _logger.debug("Error fetching %s", own_key)
+        return keys
+
+    def _schedule(self):
+        res = super(RunbotBuild, self)._schedule()
+        for build in self:
+            if not all([build.state == 'running', build.job == 'job_30_run',
+                        not build.docker_executed_commands,
+                        build.repo_id.is_travis2docker_build]):
+                continue
+            time.sleep(10)
+            build.docker_executed_commands = True
+            subprocess.call([
+                'docker', 'exec', '-d', '--user', 'root',
+                build.docker_container, '/etc/init.d/ssh', 'start'])
+            ssh_keys = self._get_ssh_keys() or ''
+            f_extra_keys = os.path.expanduser('~/.ssh/runbot_authorized_keys')
+            if os.path.isfile(f_extra_keys):
+                with open(f_extra_keys) as fobj_extra_keys:
+                    ssh_keys += "\n" + fobj_extra_keys.read()
+            ssh_keys = ssh_keys.strip(" \n")
+            if ssh_keys:
+                subprocess.call([
+                    'docker', 'exec', '-d', '--user', 'odoo',
+                    build.docker_container,
+                    "bash", "-c", "echo '%(keys)s' | tee -a '%(dir)s'" % dict(
+                        keys=ssh_keys, dir="/home/odoo/.ssh/authorized_keys"),
+                ])
+        return res
+
+    def _get_run_cmd(self):
```

Comment:
please make this `_get_docker_run_cmd` (or similar, but that's very generic and bound to clash with other stuff i guess)

Reply:
Good point!
Thanks!